include/h2o.h

```diff
@@ -1123,6 +1123,10 @@ typedef void (*h2o_forward_datagram_cb)(h2o_req_t *req, h2o_iovec_t *datagrams,
 #define H2O_SEND_SERVER_TIMING_BASIC 1
 #define H2O_SEND_SERVER_TIMING_PROXY 2
 
+#define H2O_DATAGRAM_FORMAT_NONE     0
+#define H2O_DATAGRAM_FORMAT_DRAFT03  1
+#define H2O_DATAGRAM_FORMAT_RFC      2
+
```

Comment:
These are primarily used to indicate the datagram format used in the forward direction (from client) sothat when h2o receives data to send back to the client, it uses the correct datagram format

Reply:
These might be able to be reduced to a single bitflag within `h2o_req_t` that indicates that rfc9298 datagram frames should be used. The connect handler `on_req` methods can save this flag and use it to format the datagrams accordingly.

Reply:
Thank you for you contributions and reviewing my feedback. With regard to this last suggestion, I'll defer to Kazuho.

My motivation in suggesting it is that it seems odd to set the datagram format based on the protocol header. That decision should be based solely on the H3 settings frame.

Reply:
Thanks. 

The logic here was: We will accept RFC or draft datagrams from the peer, but when it comes to sending datagrams back to the peer, we want to send it in the format (RFC or draft) that the datagrams in the 'tunnel' we received on e.g. if the client sent RFC-based datagrams we want to return RFC-based datagrams and if the client sent draft-based ones we want to return draft-based datagrams. 

The server advertises both draft and RFC in its settings to the peer. If the client will only advertise either draft or RFC, then the server only need to look at the peer settings to decide whether to use RFC or draft when returning datagrams back to the client.

I can change this behavior based on your and Kazuho's comments.

Reply:
I generally agree with the above statements. In the case of H2, the changes in this PR base the capsule format in use off of the request only. In the case of H3 we could ignore the peer settings when processing a received datagram/capsule and also simply base it off the request’s format - but the peer settings have to be verified when sending (replying) a datagram/capsule to the client. I see this PR does not yet address HTTP/1.